src/core/core.controller.js

```diff
@@ -659,14 +663,26 @@ helpers.extend(Chart.prototype, /** @lends Chart */ {
 	 */
 	transition: function(easingValue) {
 		var me = this;
+		var options = me.options || {};
+		var hoverOptions = options.hover;
+		var i, ilen;
 
-		for (var i = 0, ilen = (me.data.datasets || []).length; i < ilen; ++i) {
+		for (i = 0, ilen = (me.data.datasets || []).length; i < ilen; ++i) {
 			if (me.isDatasetVisible(i)) {
 				me.getDatasetMeta(i).controller.transition(easingValue);
 			}
 		}
 
 		me.tooltip.transition(easingValue);
+
+		if (me._lastEvent && me.animating) {
+			// If, during animation, element under mouse changes, let's react to that.
+			me.active = me.getElementsAtEventForMode(me._lastEvent, hoverOptions.mode, hoverOptions);
```

Comment:
Why not simply calling `eventHandler` but duplicating code in `handleEvent`? I think other logic such as calling the  `onHover` callback is also needed.

Reply:
That is probably right, I replaced these lines now with a call to eventHandler.